ddtrace/context.py

```diff
@@ -20,15 +20,24 @@ class Context(object):
 
     This data structure is thread-safe.
     """
-    def __init__(self):
+    def __init__(self, trace_id=None, span_id=None):
         """
         Initialize a new thread-safe ``Context``.
+
+        :param int trace_id: trace_id of parent span
+        :param int span_id: span_id of parent span
         """
         self._trace = []
         self._sampled = False
         self._finished_spans = 0
         self._current_span = None
         self._lock = threading.Lock()
+        self._parent_span_id = span_id
+        self._parent_trace_id = trace_id
+
+    def get_parent_span_ids(self):
```

Comment:
What I would like is to remove that method too, so that manual instrumentation doesn't rely in a method that brings us a little far away from the OpenTracing specification. Also, I think we cannot access individually to these attributes because the class is mutable and the current implementation requires it to be thread-safe.

I think in the overall the patch is good! let me think about it today, so I can push some changes around your code and see how it behaves! Sorry for all these interactions, but your change is really interesting and I want to make it so generic that it can be applied to other frameworks too (especially when propagating the Context in `gevent`).

Will keep you updated!